packages/playground/website/cypress/e2e/offline.cy.ts

```diff
@@ -0,0 +1,91 @@
+const assertOffline = () => {
+	return cy.wrap(window).its('navigator.onLine').should('be.false');
+};
+
+const goOffline = () => {
+	Cypress.automation('remote:debugger:protocol', {
+		command: 'Network.enable',
+	}).then(() => {
+		return Cypress.automation('remote:debugger:protocol', {
+			command: 'Network.emulateNetworkConditions',
+			params: {
+				offline: true,
+				latency: -1,
+				downloadThroughput: -1,
+				uploadThroughput: -1,
+			},
+		});
+	});
+};
+
+const goOnline = () => {
+	// https://chromedevtools.github.io/devtools-protocol/1-3/Network/#method-emulateNetworkConditions
+	Cypress.automation('remote:debugger:protocol', {
+		command: 'Network.emulateNetworkConditions',
+		params: {
+			offline: false,
+			latency: -1,
+			downloadThroughput: -1,
+			uploadThroughput: -1,
+		},
+	});
+};
+
+/**
+ * We need to intercept the static assets download because the
+ * `backfillStaticFilesRemovedFromMinifiedBuild` function is the last part of
+ * the boot process that runs when the website is online.
+ */
+const interceptWordPressStaticAssets = () => {
```

Comment:
The assumption that we're ready for an offline mode once this file is downloaded is true today, but how long will it remain true? Could Playground clearly indicate "yes, we're ready for offline"? We need it anyway for the new webapp design and it would be nice for this test to rely on a canonical information.

Reply:
> Could Playground clearly indicate "yes, we're ready for offline"?

There are a few parts that need to be completed before Playground is offline ready.

We need to ensure all offline mode assets are cached by `cacheOfflineModeAssets`, the boot needs to complete and cache the remaining files like PHP and WordPress files, at the end it needs to backfill all static assets by calling `backfillStaticFilesRemovedFromMinifiedBuild`.

The only way to be sure is to check the cache and confirm all the files are available. 
To do this, we would need to extend `assets-required-for-offline-mode.json` with the dynamically fetched assets and check the cache to see all files are there.

Reply:
@adamziel do you think that a list of files would work here or did you have another solution in mind?

Reply:
@bgrgicak I just want to avoid coupling the e2e test with offline mode’s internal logic, like „we’re offline-ready if url was downloaded”. Ideally, the e2e test would get a clear yes/no answer from the application. This could be just a body class, like `.offline-ready`.